sips/sip-x/sip-x-authentication-protocol.md

```diff
@@ -0,0 +1,397 @@
+| jti                    | string | As defined in RFC7519                                                                                                                     |
+| iat                    | string | As defined in RFC7519                                                                                                                     |
+| exp                    | string | As defined in RFC7519                                                                                                                     |
+| iss                    | string | Decentralized identifier defined in DID specification representing the user's account. See Appendix C for list of well-known DID methods. |
+| public_keys            | array  | Single item list with the public key of the signer                                                                                        |
+| domain_name            | string | The url of the application with schema.                                                                                                   |
+| manifest_uri           | string | The url of the application manifest, usually domain_name + "/manifest.json"                                                               |
+| redirect_uri           | string | The url that should receive the authentication response                                                                                   |
+| do_not_include_profile | bool   |                                                                                                                                           |
+| supports_hub_url       | bool   |                                                                                                                                           |
+| scopes                 | array  | list of strings specifying the requested access to the user's account. See Appendix A for full list of scopes                             |
+| state                  | any    | value to be echoed in the auth response                                                                                                   |
+| version                | string | must be "2.0.0"                                                                                                                           |
+
+### Verification
+
+Authenticators should verify that the request has the following properties:
+
+- expiration date (`exp`) is not in the past
+- issuance date (`iat`) is in the past
+- public keys length is 1
+- public key (`public_keys[0]`) is same as the signer's key
+- public key's stacks address is the same as the issuer
+- manifest url is same origin as the app domain
+- redirect url is same origin as the app domain
+
+## User authorization
+
+The authenticator manages the user' private keys. The protocol requires that keys are created from a deterministic wallet using [BIP-39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki).
```

Comment:
should be bip32?

Reply:
BIP32 would be sufficient. However, I would like to enforce BIP39 for usability for humans and portability?